plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/FilesTable.java

```diff
@@ -107,59 +112,65 @@ private static List<Page> buildPages(ConnectorTableMetadata tableMetadata, Table
+                pagesBuilder.appendBigint(dataFile.fileSizeInBytes());
+                if (checkNonNull(dataFile.columnSizes(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToFieldPaths(idToFieldPathMapping, dataFile.columnSizes()));
+                }
+                if (checkNonNull(dataFile.valueCounts(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToFieldPaths(idToFieldPathMapping, dataFile.valueCounts()));
+                }
+                if (checkNonNull(dataFile.nullValueCounts(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToFieldPaths(idToFieldPathMapping, dataFile.nullValueCounts()));
+                }
+                if (checkNonNull(dataFile.nanValueCounts(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharBigintMap(remapToFieldPaths(idToFieldPathMapping, dataFile.nanValueCounts()));
+                }
+                if (checkNonNull(dataFile.lowerBounds(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharVarcharMap(dataFile.lowerBounds().entrySet().stream()
+                            .filter(entry -> idToTypeMapping.containsKey(entry.getKey()))
+                            .collect(toImmutableMap(
+                                    entry -> idToFieldPathMapping.get(entry.getKey()),
+                                    entry -> Transforms.identity(idToTypeMapping.get(entry.getKey())).toHumanString(
+                                            Conversions.fromByteBuffer(idToTypeMapping.get(entry.getKey()), entry.getValue())))));
+                }
+                if (checkNonNull(dataFile.upperBounds(), pagesBuilder)) {
+                    pagesBuilder.appendVarcharVarcharMap(dataFile.upperBounds().entrySet().stream()
+                            .filter(entry -> idToTypeMapping.containsKey(entry.getKey()))
+                            .collect(toImmutableMap(
+                                    entry -> idToFieldPathMapping.get(entry.getKey()),
+                                    entry -> Transforms.identity(idToTypeMapping.get(entry.getKey())).toHumanString(
+                                            Conversions.fromByteBuffer(idToTypeMapping.get(entry.getKey()), entry.getValue())))));
+                }
+                if (checkNonNull(dataFile.keyMetadata(), pagesBuilder)) {
+                    pagesBuilder.appendVarbinary(Slices.wrappedBuffer(dataFile.keyMetadata()));
+                }
+                if (checkNonNull(dataFile.splitOffsets(), pagesBuilder)) {
+                    pagesBuilder.appendBigintArray(dataFile.splitOffsets());
+                }
+                if (checkNonNull(dataFile.equalityFieldIds(), pagesBuilder)) {
+                    pagesBuilder.appendIntegerArray(dataFile.equalityFieldIds());
+                }
+                pagesBuilder.endRow();
+            });
+        }
+        catch (IOException e) {
```

Comment:
Per other usages of `tableScan.planFiles()` I see that the exception thrown is:

```
throw new UncheckedIOException(e);
```

e.g. : https://github.com/trinodb/trino/blob/24438dd7c2f261d4e3af66f3d16e2dee9aa0ed00/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java#L239-L241

Reply:
changed